app/assets/javascripts/discourse/app/templates/activate-account.gjs

```diff
@@ -0,0 +1,140 @@
+import { service } from "@ember/service";
+import RouteTemplate from "ember-route-template";
+import DButton from "discourse/components/d-button";
+import hideApplicationHeaderButtons from "discourse/helpers/hide-application-header-buttons";
+import hideApplicationSidebar from "discourse/helpers/hide-application-sidebar";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import { wavingHandURL } from "discourse/lib/waving-hand-url";
+import i18n from "discourse-common/helpers/i18n";
+import I18n from "discourse-i18n";
+
+export default RouteTemplate(
+  class extends Component {
+    @service siteSettings;
+
+    @tracked accountActivated = false;
+    @tracked isLoading = false;
+    @tracked needsApproval = false;
+    @tracked errorMessage = null;
+
+    get siteName() {
+      return this.siteSettings.title;
+    }
+
+    get wavingHandURL() {
+      return wavingHandURL();
+    }
+
+    get translatedButtonLabel() {
+      return I18n.t("user.activate_account.continue_button", {
+        site_name: this.siteName,
+      });
+    }
+
+    @action
+    async activate() {
+      this.isLoading = true;
+
+      let hp;
+      try {
+        const response = await fetch("/session/hp", {
+          headers: {
+            Accept: "application/json",
+          },
+        });
+        hp = await response.json();
+      } catch (error) {
+        this.isLoading = false;
+        popupAjaxError(error);
+        return;
+      }
+
+      try {
+        const response = await ajax(
+          `/u/activate-account/${this.args.model.token}.json`,
+          {
+            type: "PUT",
+            data: {
+              password_confirmation: hp.value,
+              challenge: hp.challenge.split("").reverse().join(""),
+            },
+          }
+        );
+
+        if (!response.success) {
+          this.errorMessage = I18n.t("user.activate_account.already_done");
+          return;
+        }
+
+        this.accountActivated = true;
+
+        if (response.redirect_to) {
+          window.location.href = response.redirect_to;
+        } else if (response.needs_approval) {
+          this.needsApproval = true;
+        } else {
+          setTimeout(() => (window.location.href = "/"), 2000);
+        }
+      } catch (error) {
+        this.errorMessage = I18n.t("user.activate_account.already_done");
+      }
+    }
+
+    <template>
+      {{hideApplicationSidebar}}
+      {{hideApplicationHeaderButtons "search" "login" "signup"}}
+      <div id="simple-container">
+        {{#if this.errorMessage}}
+          <div class="alert alert-error">
+            {{this.errorMessage}}
+          </div>
+        {{else}}
+          <div class="activate-account">
+            <h1 class="activate-title">{{i18n
+                "user.activate_account.welcome_to"
+                site_name=this.siteName
+              }}
+              <img src={{this.wavingHandURL}} alt="" class="waving-hand" />
```

Comment:
in gjs you can use the imported function directly

```suggestion
              <img src={{(wavingHandURL)}} alt="" class="waving-hand" />
```

Reply:
but `wavingHandURL` it's a function 😬

Reply:
ohh, nicee :+1: